drivers/gles3/rasterizer_canvas_gles3.cpp

```diff
@@ -2048,16 +2048,26 @@ void RasterizerCanvasGLES3::occluder_polygon_set_cull_mode(RID p_occluder, RS::C
 void RasterizerCanvasGLES3::set_shadow_texture_size(int p_size) {
 	GLES3::Config *config = GLES3::Config::get_singleton();
 	p_size = nearest_power_of_2_templated(p_size);
-	if (p_size == state.shadow_texture_size) {
-		return;
-	}
 
 	if (p_size > config->max_texture_size) {
 		p_size = config->max_texture_size;
 		WARN_PRINT("Attempting to set CanvasItem shadow atlas size to " + itos(p_size) + " which is beyond limit of " + itos(config->max_texture_size) + "supported by hardware.");
 	}
 
+	if (p_size == state.shadow_texture_size) {
+		return;
+	}
 	state.shadow_texture_size = p_size;
+
+	if (state.shadow_fb != 0) {
+		glDeleteFramebuffers(1, &state.shadow_fb);
+		glDeleteTextures(1, &state.shadow_texture);
+		glDeleteRenderbuffers(1, &state.shadow_depth_buffer);
+		state.shadow_fb = 0;
```

Comment:
The shadow texture needs to be freed using the free_data utility to ensure that we properly track resources
```suggestion
		glDeleteFramebuffers(1, &state.shadow_fb);
		GLES3::Utilities::get_singleton()->texture_free_data(state.shadow_texture);
		glDeleteRenderbuffers(1, &state.shadow_depth_buffer);
		state.shadow_fb = 0;
```